cirq/google/sim/xmon_simulator.py

```diff
@@ -166,6 +167,35 @@ def __init__(self, options: XmonOptions = None) -> None:
         """
         self.options = options or XmonOptions()
 
+    def wavefunction(self, state, decimals=2):
+        """
+        Returns the wavefunction as a string in Dirac notation.
+
+        For example:
+        circuit = cirq.Circuit.from_ops(cirq.H(Q1))
+        simulator = cirq.google.XmonSimulator()
+        result = simulator.simulate(circuit)
+
+        print(wavefunction(result.final_state)) -> -0.71j|0> + -0.71j|1>
+        """
+        wvf = state.final_state
+        perm_list = ["".join(seq) for seq in itertools.product(
+            "01", repeat=int(np.log2(len(wvf))))]
+        wvf_string = ""
+
+        for x in range(0, len(perm_list)):
+            rounded_elem = np.around(wvf[x], decimals) + np.complex(0, 0)
+            rounded_elem = np.around(rounded_elem, decimals)
```

Comment:
I don't quite understand what the np.complex is doing here and why there are two arounds.

Reply:
Yeah. These lines are suspect. I should have left a comment or two justifying them. This was the only way I could think on how to handle `-0.`. `circuit = cirq.Circuit.from_ops(cirq.H(Q1))` prints `(-0-0.71j)|0⟩ + -0.71j|1⟩`. That negative 0 in front of `|0⟩ ` doesn't look good. To not print that `-0`, I use the fact that `-0 + 0 = 0` , so that `0` is not printed. The line doesn't need an np.complex, so I should change that to `rounded_elem = np.around(wvf[x], decimals) + 0.` The problem is that this line undos the rounding for some reason; making rounded_elem `-0.7099999785423279j|0⟩ + -0.7099999785423279j|1⟩`. That's why I call `round` again.

There has to be a better way! Just haven't thought of it.

Reply:
Maybe `rounded_elem = np.around(wvf[x], decimals) + 0.0` will eliminate the unnecessary round procedure call.